mypy/checkmember.py

```diff
@@ -550,7 +550,8 @@ def analyze_var(name: str,
             return mx.chk.handle_partial_var_type(typ, mx.is_lvalue, var, mx.context)
         if mx.is_lvalue and var.is_property and not var.is_settable_property:
             # TODO allow setting attributes in subclass (although it is probably an error)
-            mx.msg.read_only_property(name, itype.type, mx.context)
+            if info.names.get('__setattr__') is None:
```

Comment:
how about `if '__setattr__' not in info.names:`

Reply:
The best thing would be to use `info.get`, since that also looks at base classes that might define `__setattr__`.